root/usr/bin/discovery-fetch-extensions

```diff
@@ -70,7 +70,7 @@ end
 
 # Write out environment files for subsequent paths and systemd units to use
 File.open('/etc/default/discovery','w') do |f|
-  f.write("# EnvironmentFile generated by discovery-setup.service\n")
+  f.write("# EnvironmentFile generated by discovery-fetch-extensions.service\n")
   f.write(env_append('PATH','/opt/extension/bin'))
   f.write(env_append('FACTERLIB','/usr/share/fdi/facts:/opt/extension/facts'))
   f.write(env_append('LD_LIBRARY_PATH','/opt/extension/lib'))
```

Comment:
As long as the script (and its includes/requires) do not alter the systemd set environment variables /etc/default/discovery seems to be pretty static and it should be save to drop these statements.

Do we still need ruby for this script afterwards? If not I suggest:

    #!/bin/bash
    
    # Use a commandline option, fdi.zips=zip1.zip,zip2.zip,etc and
    # get the zips from the TFTP server. Try zipserver passed on kernel
    # cmdline first and fall back to DHCP next-server passed in through
    # ENV
    
    local extdir="/opt/extension"
    
    source /usr/share/fdi/commonfunc.sh
    exportKCL
    
    for zip in ${FDI_ZIPS//,/ }
    do
      local tempfile="$(mktemp)"
      curl --fail --silent "tftp://${ZIP_SERVER}/${zip}" -o "${tempfile}"
      local curl_rc="$?"
      if [ "${curl_rc}" -ne 0 ]
      then
        echo "CURL error downloading ${zip} via tftp: ${curl_rc}"
        echo "Problem downloading zipfile ${zip}" >&2
      else
        unzip "${tempfile}" -d "${extdir}"
      fi
      rm "${tempfile}"
    done